azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2017-09-30/containerservice"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmManagedCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedClusterCreate,
+		Read:   resourceArmManagedClusterRead,
+		Update: resourceArmManagedClusterCreate,
+		Delete: resourceArmManagedClusterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"dns_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"kubernetes_version": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
I believe this should be required?

Reply:
`kubernetes_version` is optional, default is `1.7.9` right now, but can be changed up to `1.9.1`.